Model section split into 'preliminary' and 'modifications' (#85)
MrLongbottom/P10-Article: sections/plate_notation.tex

\section{Our Topic Models}\label{sec:plate_notation}

> In this section we will detail the modifications that we have implemented to the existing models, outlined in \autoref{sec:preliminaries}.

Reviewer: ```suggestion
In this section, we detail the modifications that we have implemented to the existing models, outlined in \autoref{sec:preliminaries}.
```

> We will be presenting three models, one for each of the metadata fields detailed in \autoref{sec:dataset}.

Reviewer: ```suggestion
We present three models, one for each of the metadata fields detailed in \autoref{sec:dataset}.
```

> Two of the models: the Author and Category metadata models, are based on the Author-topic \gls{lda} explained in \autoref{subsec:auth_prelim}.

Reviewer: ```suggestion
Two of the models: the Author and Category metadata models are based on the Author-topic \gls{lda} explained in \autoref{subsec:auth_prelim}.
```

Lastly the Taxonomy metadata model is based on \gls{pam}, which is explained in \autoref{subsec:pachinko_prelim}.

\subsection{Author-Topic \gls{lda} and Category-Topic \gls{lda}}
We model both of the metadata fields 'Author' and 'Category' similarly to the model by \citet{author_topic_2012} presented in \autoref{subsec:auth_prelim}.
The category model builds on the assumption that categories of the articles were chosen based on the content of the finished article and that local newspapers have their own unique topic preferences.

> We find this model to be generally applicable to most metadata information assuming that the metadata is either chosen based on the text of the documents (as with the category metadata) or that it has some impact on the text of the document (as with the author metadata).

Reviewer 2: ```suggestion
We find this model to be generally applicable to most metadata information, assuming that the metadata is either chosen based on the text of the documents (as with the category metadata) or that it has some impact on the text of the document (as with the author metadata).
```

> We find this model to be generally applicable to most metadata information assuming that the metadata is either chosen based on the text of the documents (as with the category metadata) or that it has some impact of the text of the document (as with the author metadata).

Reviewer: ```suggestion
We find this model to be generally applicable to most metadata information assuming that the metadata is either chosen based on the text of the documents (as with the category metadata) or that it has some impact on the text of the document (as with the author metadata).
```

> For our category-topic model and our own author-topic model, each document $d$ is associated with only one category $c_d$ from a set of categories $C$ and only one author $a_d$ from a set of authors $A$, rather than a vector.

Reviewer: ```suggestion
For our category-topic model and our author-topic model, each document $d$ is associated with only one category $c_d$ from a set of categories $C$ and only one author $a_d$ from a set of authors $A$, rather than a vector.
```
måske ref til author-topic

\subsection{Pachinko Allocation}
In order to handle the hierarchical structure of the taxonomy metadata field, we use a hierarchical topic model, namely the \acrfull{pam} from \citet{li2006pachinko}.

In this paper, we construct a five-level \gls{pam}, following the format presented by \citet{li2006pachinko}.
The \gls{dag} structure used in this paper is in \autoref{fig:pachinko_dag}, where each layer has been named.
The plate notation for this model is shown in \autoref{fig:pachinko}

Some layers in our \gls{dag} structure are duplicates of layers from the taxonomy field within our dataset.

> This allows us to have the some nodes represent a topic based on a specific taxonomy.

Reviewer: ```suggestion
This allows us to have some nodes represent a topic based on a specific taxonomy.
```

To make the algorithm construct the topics to be based on our taxonomies, we introduce a novel locking mechanism for the Gibbs sampler which we use to run \gls{pam}.
This mechanism is discussed further in \autoref{subsec:mod_pachinko}.

The first layer of the \gls{dag} is the root layer.
The last layer is the word layer consisting of one node for each word in the vocabulary of our corpus.
The second and third layers will be constructed based on the entries of the first two positions in our taxonomy metadata, meaning there will be one node for each unique sub-taxonomy entry that is in the first or second position in the taxonomy sequence (e.g. "STEDER" and "Danmark", which is taken from "STEDER/Danmark/Aalborg", but not "Aalborg" since it is in the third position).

> We only use the first two layers for this, as these are among the most informative, and because introducing even more layers would slow down the training significantly, since the probability of all possible combinations of topics needs to be sampled for every word during training. \vejleder[inline]{can you back this up with experiments or hard evidence?}

Author: ```suggestion
We only use the first two layers for this, as these are among the most informative, and because introducing even more layers would slow down the training significantly, since the probability of all possible combinations of topics needs to be sampled for every word during training.
From our experiments, the training time for 50 epochs, increases from 12 hours to 130 hours between four and five level pachinko.
```

> We only use the first two layers for this, as these are among the most informative, and because introducing even more layers would slow down the training significantly, since the probability of all possible combinations of topics needs to be sampled for every word during training. \vejleder[inline]{can you back this up with experiments or hard evidence?}

Reviewer: måske kom med 12 timer vs 130 timer I run time ?

The first layer of the \gls{dag} is the root layer.
The last layer is the word layer consisting of one node for each word in the vocabulary of our corpus.
The second and third layers will be constructed based on the entries of the first two positions in our taxonomy metadata, meaning there will be one node for each unique sub-taxonomy entry that is in the first or second position in the taxonomy sequence (e.g. "STEDER" and "Danmark", which is taken from "STEDER/Danmark/Aalborg", but not "Aalborg" since it is in the third position).
We only use the first two layers for this, as these are among the most informative, and because introducing even more layers would slow down the training significantly, since the probability of all possible combinations of topics needs to be sampled for every word during training.
From our experiments, the training time for 50 epochs, increases from 12 hours to 130 hours between four and five level pachinko.
The fourth layer consists of $K = 90$ 'blank' topics, as with the other models we present in this paper.
This layer is added so that the model can construct its own topics based on the higher-level topics learned from our taxonomy metadata.

> We use Gibbs sampling for performing inference.

Reviewer 2: Det kan være, som jeg forstod det fra Thomas, at det vil give mere mening at have denne Gibbs sampling beskrivelse sammen med den i bilaget, og så bare her referere til appendix. Så sparer vi også lidt plads.

Author: for min skyld gerne

Reviewer 2: Jeg kan se at "Pachinko Implementation" i appendix også beskrive skridtene i Pachinkos Gibbs sampling, så det er også en mulighed at flytte den derned. Ved dog ikke hvad vil passe bedst.
